tests/service/test_rest_api.py

```diff
@@ -23,7 +23,7 @@ class MyModel(BaseModel):
 
     plan = Plan(name="my-plan", model=MyModel)
 
-    handler.context.plans = {"my-plan": plan}
+    handler._context.plans = {"my-plan": plan}
```

Comment:
I know I made this change, which makes this comment seem hypocritical; but I think tests shouldn't use the underlying ._context and other _ attributes if they can avoid it. So this is just a comment to say; please review everywhere where tests do something like handler._ ... and see if they can't use the facade methods instead.

That's all. Otherwise looking good.

Reply:
Thanks - yes I was wondering about this too and I do agree.
I'm a bit unsure how to actually achieve this though. E.g. how to specify specific plans and devices used in the tests without accessing these fields directly. In the normal code flow they are loaded based on the `ApplicationConfig` but the tests at the moment don't concern themselves with that.

Reply:
Agree with Rose, is there a way we can restructure this test?

Reply:
Due to the complexity of the test changes required I think we should handle this in another PR/issue.

The tests will need to change with the next PR (that actually adds the subprocess) too.